### PR TITLE
Pass environment variable culture from Visual Studio/VSCode to command line processes

### PR DESCRIPTION
**Pull request by ocallesp**
This pull request aims to propagate the Culture and UICulture settings from Visual Studio and VS Code to the child processes responsible for executing the code (kernels).

Three new unit tests:
- Culture_Is_Set_Based_On_Environment_Variable
- UICulture_Is_Set_Based_On_Environment_Variable
- Kernel_Execute_Code_Using_Culture_From_Parent_Thread_That_Created_It

### CODE REVIEW

File: src/dotnet-interactive/Program.cs [hunk 1]
```diff
@@ -30,6 +31,16 @@ public class Program
     public static async Task<int> Main(string[] args)
     {
         Console.OutputEncoding = Encoding.UTF8;
+
+        var culture = Environment.GetEnvironmentVariable("DOTNET_CLI_CULTURE");
+        var uiLanguage = Environment.GetEnvironmentVariable("DOTNET_CLI_UI_LANGUAGE");
+
+        if (!string.IsNullOrWhiteSpace(culture) && !string.IsNullOrWhiteSpace(uiLanguage))
```
Comment (jonsequitur) [inline src/dotnet-interactive/Program.cs:38]: These can be set independently so splitting this into separate `if` blocks would be a good idea.

File: src/dotnet-interactive/Connection/StdIoKernelConnector.cs [hunk 2]
```diff
@@ -99,7 +100,9 @@ public async Task<ProxyKernel> CreateRootProxyKernelAsync()
                     {
                         ["DOTNET_INTERACTIVE_SKIP_FIRST_TIME_EXPERIENCE"]  = "1",
                         ["DOTNET_SKIP_FIRST_TIME_EXPERIENCE"]  = "1",
-                        ["DOTNET_DbgEnableMiniDump"] = "0" // https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dumps
+                        ["DOTNET_DbgEnableMiniDump"] = "0", // https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dumps
+                        ["DOTNET_CLI_UI_LANGUAGE"] = GetCurrentUICulture(),
+                        ["DOTNET_CLI_CULTURE"] = GetCurrentCulture()
```
Comment (jonsequitur) [inline src/dotnet-interactive/Connection/StdIoKernelConnector.cs:105]: A similar change will also need to be made in the Polyglot Notebooks extension when it launches the `dotnet-interactive` process, here: 

https://github.com/dotnet/interactive/blob/main/src/polyglot-notebooks-vscode-common/src/extension.ts#L130-L157

Comment (ocallesp) [reply src/dotnet-interactive/Connection/StdIoKernelConnector.cs:105]: done

File: src/dotnet-interactive/Program.cs (hunk 1 above)
Comment (ocallesp) [reply src/dotnet-interactive/Program.cs]: changed

File: src/dotnet-interactive.Tests/StdIoBehaviorTests.cs [hunk 3]
```diff
@@ -12,6 +15,106 @@
 
 public class StdIoBehaviorTests
 {
+    [Fact]
+    public async Task Pass_Culture_to_stdio_process()
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/StdIoBehaviorTests.cs:19]: Let's chat about how to make this test a little simpler.

File: src/polyglot-notebooks-vscode-common/src/extension.ts [hunk 4]
```diff
@@ -140,7 +140,8 @@ export async function activate(context: vscode.ExtensionContext) {
 
         const workspaceFolderUris = vscode.workspace.workspaceFolders?.map(folder => folder.uri) || [];
         const workingDirectory = getWorkingDirectoryForNotebook(notebookUri, workspaceFolderUris, fallbackWorkingDirectory);
-        const environmentVariables = polyglotConfig.get<{ [key: string]: string }>('kernelEnvironmentVariables');
+        const environmentVariables = { ...polyglotConfig.get<{ [key: string]: string }>('kernelEnvironmentVariables'), 'DOTNET_CLI_CULTURE': getCurrentCulture(), 'DOTNET_CLI_UI_LANGUAGE': getCurrentCulture() };
+
```
Comment (jonsequitur) [inline src/polyglot-notebooks-vscode-common/src/extension.ts:144]: A similar change will have to be made in a few places, unfortunately:

image

@colombod We should discuss whether we can reduce some duplication here.

Comment (ocallesp) [reply src/polyglot-notebooks-vscode-common/src/extension.ts:144]: There is only one place.

Extension.ts is a link and it appears at these different places

File: src/dotnet-interactive.Tests/StdIoBehaviorTests.cs (hunk 3 above)
Comment (ocallesp) [reply src/dotnet-interactive.Tests/StdIoBehaviorTests.cs]: done

**Review by jonsequitur — APPROVED** [review]

**Review by shyamnamboodiripad — APPROVED** [review]

**Pull request merged**